lib/stm32/common/st_usbfs_core.c

```diff
@@ -37,44 +37,56 @@ void st_usbfs_set_address(usbd_device *dev, uint8_t addr)
 	SET_REG(USB_DADDR_REG, (addr & USB_DADDR_ADDR) | USB_DADDR_EF);
 }
 
-/**
- * Set the receive buffer size for a given USB endpoint.
- *
- * @param dev the usb device handle returned from @ref usbd_init
- * @param ep Index of endpoint to configure.
- * @param size Size in bytes of the RX buffer. Legal sizes : {2,4,6...62}; {64,96,128...992}.
- * @returns (uint16) Actual size set
+/*
+ * Convert from raw byte count to special encoded value suitable for writing
+ * to USB_COUNTn_RX registers, which consists of NUM_BLOCK[4:0] and BLSIZE.
+ * Returns the corresponding actual size in *real_size.
  */
-uint16_t st_usbfs_set_ep_rx_bufsize(usbd_device *dev, uint8_t ep, uint32_t size)
+static uint32_t convert_bufsize(uint32_t size, uint16_t *real_size)
 {
-	uint16_t realsize;
-	(void)dev;
+	uint32_t encoded;
 	/*
 	 * Writes USB_COUNTn_RX reg fields : bits <14:10> are NUM_BLOCK; bit 15 is BL_SIZE
 	 * - When (size <= 62), BL_SIZE is set to 0 and NUM_BLOCK set to (size / 2).
 	 * - When (size > 62), BL_SIZE is set to 1 and NUM_BLOCK=((size / 32) - 1).
 	 *
 	 * This algo rounds to the next largest legal buffer size, except 0. Examples:
-	 *	size =>	BL_SIZE, NUM_BLOCK	=> Actual bufsize
-	 *	0		0		0			??? "Not allowed" according to RM0091, RM0008
-	 *	1		0		1			2
-	 *	61		0		31			62
-	 *	63		1		1			64
+	 *	size =>		BL_SIZE,	NUM_BLOCK	=> Actual bufsize
+	 *	0		0		0		??? "Not allowed" according to RM0091, RM0008
+	 *	1		0		1		2
+	 *	61		0		31		62
+	 *	63		1		1		64
 	 */
 	if (size > 62) {
 		/* Round up, div by 32 and sub 1 == (size + 31)/32 - 1 == (size-1)/32)*/
-		size = ((size - 1) >> 5) & 0x1F;
-		realsize = (size + 1) << 5;
+		encoded = ((size - 1) >> 5) & 0x1F;
+		*real_size = (encoded + 1) << 5;
 		/* Set BL_SIZE bit (no macro for this) */
-		size |= (1<<5);
+		encoded |= (1<<5);
 	} else {
 		/* round up and div by 2 */
-		size = (size + 1) >> 1;
-		realsize = size << 1;
+		encoded = (size + 1) >> 1;
+		*real_size = encoded << 1;
 	}
+	return encoded;
+}
+
+/**
+ * Set the receive buffer size for a given USB endpoint.
+ *
+ * @param dev the usb device handle returned from @ref usbd_init
+ * @param ep Index of endpoint to configure.
+ * @param size Size in bytes of the RX buffer. Legal sizes : {2,4,6...62}; {64,96,128...992}.
+ * @returns (uint16) Actual size set
+ */
+uint16_t st_usbfs_set_ep_rx_bufsize(usbd_device *dev, uint8_t ep, uint32_t size)
+{
+	uint16_t real_size;
+
+	(void)dev;
 	/* write to the BL_SIZE and NUM_BLOCK fields */
-	USB_SET_EP_RX_COUNT(ep, size << 10);
-	return realsize;
+	USB_SET_EP_RX_COUNT(ep, convert_bufsize(size, &real_size) << 10);
+	return real_size;
```

Comment:
What's actually even chantged here then?  Please use the commit message to explain why this has to change as well, it appears to be unrelated to "add double buffering for isochronous" and the whitespace changes above make it very hard to actually follow

Reply:
This part was part of _set_ep_rx_bufsize but the calculation was factored out into convert_bufsize(). Maybe the diff output makes this unclear. If you look at the new code it should be simple.

Reply:
But yes, we can mention in the commit message that the RX buffer size was factored out. Maybe there's a better name too. encode_rx_bufsize() ?